src/__tests__/restLink.ts

```diff
@@ -2497,6 +2695,111 @@ describe('Query options', () => {
       ]);
     });
   });
+  describe('body', () => {
+    it('prioritizes fetchOptions body over directive body', async () => {
```

Comment:
Where does `fetchOptions body` come from? What is it for?

Reply:
There are a lot of interplaying features, and I'm not super comfortable with blindly accepting a body via a context parameter without understanding the spec.

The abortcontroller is interesting and makes sense to me, but I don't think the body should be passed through?

Reply:
I don't think the `body` option is necessarily the most useful thing, but I included it for completeness. I was copying the behavior in the [HTTP Link](https://www.apollographql.com/docs/react/api/link/apollo-link-http/#fetchoptions-1), which simply passes all [fetch options](https://developer.mozilla.org/en-US/docs/Web/API/fetch#parameters) to the fetch function. I think it makes sense to include this option so no one tries to use it and gets confused by it not working (which is the state I was in when I tried to use an abort controller). It should not break any existing behavior.

Reply:
The problem is that we can't guarantee other Links aren't accidentally injecting a body, and if other links do inject a body, we can't parse the results or otherwise handle it.

I get why AbortController makes sense, but if we use the body, we're actually building apollo-link-rest into a really inefficient clone of HTTPLink that doesn't do the right thing?

Maybe I'm confused as to how this body is used normally and when it would be expected to be passed to this link.